Products/Zuul/facades/devicefacade.py

```diff
@@ -953,9 +954,12 @@ def getComponentGraphs(self, uid, meta_type, graphId, allOnSame=False):
             return []
 
         if allOnSame:
-            # ZEN-26498 identify by name to match individual charts
+            # ZEN-26498 ZEN-27126 get docker legends to match individual graphs
             for comp in components:
-                comp.id = comp.name()
+                if (hasattr(comp,'docker_host')):
```

Comment:
It may not be a monkey patch, but I think talking to solutions may help understand why the object doesn't behave the same as others and a change can be made to the zenpack.